moshi/src/main/java/com/squareup/moshi/ClassJsonAdapter.java

```diff
@@ -75,6 +75,14 @@
       if (Modifier.isAbstract(rawType.getModifiers())) {
         throw new IllegalArgumentException("Cannot serialize abstract class " + rawType.getName());
       }
+      for (Annotation annotation : rawType.getDeclaredAnnotations()) {
+        if ("kotlin.Metadata".equals(annotation.annotationType().getName())) {
+          throw new IllegalArgumentException("Cannot serialize Kotlin type " + rawType.getName()
+              + ". Reflective serialization of Kotlin classes without using kotlin-reflect can "
+              + "have undefined and unexpected behavior. Please use KotlinJsonAdapter from the "
```

Comment:
nit: instead of "can have undefined and unexpected behavior," how about "has undefined and unexpected behavior?"

Reply:
Done ca79cce